aten/src/ATen/core/jit_type.h

```diff
@@ -1543,7 +1524,30 @@ struct CAFFE2_API ClassType : public NamedType {
     return parameterSlots_->at(slot);
   }
 
+  size_t addAttribute(
+      const std::string& name,
+      TypePtr type,
+      bool is_parameter = false);
+
+  void addMethod(Function* method);
+  Function* getMethod(const std::string& name) const;
+
+#ifndef DECOUPLE_JIT
```

Comment:
Let's not change the API of this class based on a flag. My suggestion is only to change the implementation of these methods to throw exceptions. This prevents having to make modifications in as many places when adding a method that cannot be enabled for mobile.

Reply:
Have removed the #ifndef in this header file and added implementations in the cpp file.